pkg/daemon/ceph/osd/volume.go

```diff
@@ -613,20 +633,49 @@ func (a *OsdAgent) initializeDevices(context *clusterd.Context, devices *DeviceO
 			"json",
 		}...)
 
-		cvOut, err := context.Executor.ExecuteCommandWithCombinedOutput(baseCommand, reportArgs...)
-		if err != nil {
-			return errors.Wrapf(err, "failed ceph-volume json report: %s", cvOut) // fail return here as validation provided by ceph-volume
-		}
+		cephVersion := a.clusterInfo.CephVersion
 
-		logger.Debugf("ceph-volume report: %+v", cvOut)
+		// ceph version v14.2.13 ~ v14.2.16 changes output of `lvm batch --prepare --report`
+		// use previous logic if ceph version does not fall into this range
+		if !isNewStyledLvmBatch(cephVersion) {
 
-		var cvReport cephVolReport
-		if err = json.Unmarshal([]byte(cvOut), &cvReport); err != nil {
-			return errors.Wrap(err, "failed to unmarshal ceph-volume report json")
-		}
+			cvOut, err := context.Executor.ExecuteCommandWithCombinedOutput(baseCommand, reportArgs...)
```

Comment:
This command is the same for both types of reports, the output is just different, right? Can you factor out this command before the version check?

Reply:
Sadly it is not the same... new format will print something into stderr which is problematic if using `ExecuteCommandWithCombinedOutput` which seems collecting stdout + stderr....

Reply:
But actually it might be ok to change both of them to `ExecuteCommandWithOutput` ....